apps/shared/y4m.c

```diff
@@ -361,42 +361,23 @@ avifBool y4mRead(const char * inputFilename, avifImage * avif, avifAppSourceTimi
         goto cleanup;
     }
 
-    avifPixelFormatInfo formatInfo;
-    avifGetPixelFormatInfo(avif->yuvFormat, &formatInfo);
-
-    const int planeCount = formatInfo.monochrome ? 1 : AVIF_PLANE_COUNT_YUV;
-    for (int plane = 0; plane < planeCount; ++plane) {
-        uint32_t planeWidth = (plane > 0) ? ((avif->width + formatInfo.chromaShiftX) >> formatInfo.chromaShiftX) : avif->width;
-        uint32_t planeHeight = (plane > 0) ? ((avif->height + formatInfo.chromaShiftY) >> formatInfo.chromaShiftY) : avif->height;
-        uint32_t rowBytes = planeWidth << (avif->depth > 8);
-        uint8_t * row = avif->yuvPlanes[plane];
+    const int lastPlane = frame.hasAlpha ? AVIF_CHAN_A : AVIF_CHAN_V;
+    for (int plane = AVIF_CHAN_Y; plane <= lastPlane; ++plane) {
+        uint32_t planeHeight = avifImagePlaneHeight(avif, plane);
+        uint32_t sampleBytes = avifImagePlaneWidth(avif, plane) << (avif->depth > 8);
```

Comment:
Nit/optional: `sampleBytes` could be misinterpreted to mean the sample size in bytes, i.e., 1 or 2 bytes. We could name this variable `widthBytes` or `planeWidthBytes` (i.e., the width in bytes).

UPDATE: You named an equivalent variable `width_byte_count` in tests/gtest/avifincrtest_helpers.cc. That name is similar to my suggestion.